src/components/Exchange/SwapBox.js

```diff
@@ -1642,10 +1643,14 @@ export default function SwapBox(props) {
   };
 
   const onSwapOptionChange = (opt) => {
+    // Keep same fromToken when toggling swap option
+    const _swapOption = swapOption;
+    const prevTokenAddress = tokenSelection[_swapOption].from;
+    const updatedTokenSelection = JSON.parse(JSON.stringify(tokenSelection));
+    updatedTokenSelection[opt].from = prevTokenAddress;
```

Comment:
do we need to stringify then parse? could we just do { ...tokenSelection } if you wanted to shallow clone?

Reply:
yes we do, as the value is stored not only in state but in localStorage